### PR TITLE
feat(julia): update parser and queries

### PR DESCRIPTION
**Pull request by clason**
Queries synced with upstream (preserving obvious local additions, but there could be regressions; most but not all of the diff are reshuffles).

### CODE REVIEW

File: queries/julia/highlights.scm [hunk 1]
```diff
@@ -194,124 +142,134 @@
...
+  "."
+  "..."
+  "::"
+] @punctuation.special
 
 [
   ","
-  "."
   ";"
-  "::"
```
Comment (ribru17) [inline queries/julia/highlights.scm:309]: I feel like this should be left as `@punctuation.delimiter`, looks like most other languages use that highlight for this double colon thing. Unless there was a specific reason

**Review by ribru17 — APPROVED** [review]

**Comment by ribru17** [comment]
Generating the parser gives me several warnings (and takes a long time) btw:

```
Warning: rule _pair_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _lazy_or_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _lazy_and_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _pipe_right_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _pipe_left_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _rational_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
Warning: rule _tilde_operator is just a `seq` or `choice` rule with a single element. This is unnecessary.
```

will make a PR

**Comment by clason** [comment]
> Generating the parser gives me several warnings (and takes a long time) btw:

Yes, I know.

> will make a PR. Also the built parser is about 6 mb with many states 😬

Yes, I know, but that was the case already before the update. The problem (besides over-emphasis on reproducing the official syntax) is the huge list of Unicode symbols in the grammar.

File: queries/julia/highlights.scm (hunk 1 above)
Comment (clason) [reply queries/julia/highlights.scm:309]: Upstream just doesn't like our queries. 🤷

**Pull request merged**